GitCommands/RevisionReader.cs

```diff
@@ -28,6 +28,7 @@ public enum RefFilterOptions
 
     public sealed class RevisionReader : IDisposable
     {
+        private const string EndOfBody = "\x2"; // STX control char.  Nobody will ever type character 2 in the body.
```

Comment:
Hmm. I don't think you can make that assumption unfortunately. It seems reasonable for ASCII, but in UTF-8 or other encodings it's more likely to pop up as part of a multi-byte character.

Reply:
> Run it. It does the conversion. It most certainly does work.

I'm sure it works on the GE repo, but for users who write commit messages in non Latin languages, `\x2` is likely to pop up and cause this to fail.

It took me a while to conclude that a null byte was safe. Because `\0` is special in C-style strings, it's avoided in encodings (at least as far as I could determine). That's not the case for `x2`, so we need a different solution. Thinking...

Reply:
Oh crap. Dang it. Good catch.

Reply:
You could use a Guid instead... :)

Reply:
I don't want a crazy long value the nulls are the better solution.

Reply:
Sure, but if I read the code correctly, the nulls are currently used to separate the individual revisions. This makes the parser much simpler, but the corresponding refactoring (https://github.com/gitextensions/gitextensions/commit/1f6b0cacc8bc78be076b4e706e5aa030b6d36ea0) broke (part of) rename following in the first place.

I guess you could revert back to a similar parsing logic like before and use a ["(__BEGIN_COMMIT__)>" marker](https://github.com/gitextensions/gitextensions/commit/1f6b0cacc8bc78be076b4e706e5aa030b6d36ea0#diff-746e3f7b1adff7195f50bea04896ccc2L34)... I'd find it simpler, though, to use just separate commit body and file name by a pattern extremely unlikely to occur within a commit. Guid was only an example, maybe a binary pattern such as `de ad be ef` or something instead?

(About the length: the commits are already quite long anyway, I don't think it would matter to add a separator string the length of a Guid. You could also binary-encode the Guid to make it shorter.)

Reply:
Please consider how we can test the changes

Reply:
Most definitely.

Reply:
Change in master to only get filenames in the initial `git-log --follow` as it is virtually impossible to parse filenames in the "normal" `git-log`.